openedx/core/djangoapps/programs/utils.py

```diff
@@ -242,6 +247,14 @@ def _is_program_complete(self, program):
         """
         return all(self._is_course_complete(course) for course in program['courses'])
 
+    def get_course_image(self, course):
+        course_image = course.get('image')
+        if not course_image or not course_image.get('src'):
+            log.warning('Image source for course [%s] does not exist', course['key'])
```

Comment:
I think we should log an error here instead of warning. It would be bad to see this condition happening